### PR TITLE
feat(components): input

### PR DESCRIPTION
**Pull request by dahaca**
## Description

This one is big and includes changes relevant for `Select` and `Textarea`

* prop refactor – many unnecessary props have been removed for a simpler component API.
* simplified input addons usage
* `BottomText` as a separate component
* update `Input` design

## Related Issue

Resolve https://github.com/trezor/trezor-suite/issues/8997

## Screenshots:

*see storybook

### CODE REVIEW

**Comment by AdamSchinzel** [comment]
When I checked the design with Figma, I would some stuff:
1. `Textarea` is missing the dragger icon in the bottom right corner and label is not inside the component itself
1. When `Select` is searchable and has an option selected already, the cursor is not right after the text (see screen shot, probably the cursor shouldn't be there at all if option is selected)
B2CC1C95-0175-4B56-B406-72592B6BEAFE_4_5005_c
1. Height of `Select` is changing when I toggle disable to true (also applies to `withDropdownIndicator` set to false)
1. When `isClean` on `Select` is set to true, the dropdown still has 100% width (so `isSearchable` also looks weird, not sure if this is just inside Storybook bug)

File: packages/components/src/components/form/BottomText.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import React from 'react';
```
Comment (AdamSchinzel) [inline packages/components/src/components/form/BottomText.tsx]: nit: can you namely import `ReactNode` please

File: packages/components/src/config/motion.ts [hunk 2]
```diff
@@ -17,9 +17,20 @@ export const motionAnimation = {
     },
 };
 
+// TODO: move to theme package
```
Comment (AdamSchinzel) [inline packages/components/src/config/motion.ts:20]: Is it for later or todo for this PR?

File: packages/components/src/support/types/index.ts [hunk 3]
```diff
@@ -6,9 +6,9 @@ import { THEME } from '../../config/colors';
 
 export type TextAlign = 'left' | 'center' | 'right';
 
-export type InputState = 'success' | 'warning' | 'error';
+export type InputState = 'success' | 'warning' | 'error'; // TODO: remove success
```
Comment (AdamSchinzel) [inline packages/components/src/support/types/index.ts:9]: Is it for later or todo for this PR?

**Comment by dahaca** [comment]
@AdamSchinzel I will make the fixes when I move to `Select` and `Textarea`. While I did make changes to those components here, I tried to spend the least possible amount of time.

File: packages/suite/src/views/wallet/send/components/Outputs/components/Amount/index.tsx [hunk 4]
```diff
@@ -246,7 +261,7 @@ export const Amount = ({ output, outputId }: AmountProps) => {
                                 )}
                             </Label>
                         }
-                        bottomText={bottomText}
+                        bottomText={bottomText || null}
```
Comment (AdamSchinzel) [inline packages/suite/src/views/wallet/send/components/Outputs/components/Amount/index.tsx:264]: Why did you add `|| null`?

**Review by AdamSchinzel — COMMENTED** [review]
Would be also nice if you use `StoryObj` because it can nicely detect the controls for props from TypeScript. Right now it's missing some controls such as input state so I can't for example preview easily error state.

File: packages/components/src/components/form/SelectBar/SelectBar.tsx [hunk 5]
```diff
@@ -27,7 +28,7 @@ const Options = styled.div<{ isInLine: boolean }>`
     display: flex;
     flex: ${({ isInLine }) => isInLine && '1'};
     padding: 0 4px;
```
Comment (AdamSchinzel) [inline packages/components/src/components/form/SelectBar/SelectBar.tsx:30]: Is there some rule where we use these values from `spacingsPx` and where we just put them like this. Saw multiple places where the value for padding, margin, text-spacing etc. can be used from `spacingsPx`.

File: packages/components/src/config/motion.ts (hunk 2 above)
Comment (dahaca) [reply packages/components/src/config/motion.ts:20]: for later

File: packages/components/src/support/types/index.ts (hunk 3 above)
Comment (dahaca) [reply packages/components/src/support/types/index.ts:9]: Later, not an crucial refactor

File: packages/suite/src/views/wallet/send/components/Outputs/components/Amount/index.tsx (hunk 4 above)
Comment (dahaca) [reply packages/suite/src/views/wallet/send/components/Outputs/components/Amount/index.tsx:264]: https://github.com/trezor/trezor-suite/blob/04a0e2ce8aeeceb8377b5a8425089b24b835c35c/packages/components/src/components/form/Input/Input.tsx#L100

File: packages/components/src/components/form/SelectBar/SelectBar.tsx (hunk 5 above)
Comment (dahaca) [reply packages/components/src/components/form/SelectBar/SelectBar.tsx:30]: `SelectBar` is updated in another PR :) In general, I would always use the `spacings` tokens, unless the designs use custom spacings. Which is rarely the case with the new designs.

File: packages/components/src/components/form/BottomText.tsx (hunk 1 above)
Comment (dahaca) [reply packages/components/src/components/form/BottomText.tsx]: 2d4f3f9bd

File: packages/suite/src/components/suite/PinInput/components/InputPin/index.tsx [hunk 6]
```diff
@@ -3,15 +3,17 @@ import styled from 'styled-components';
...
-    min-width: 256px;
-    height: 54px;
-    font-weight: ${variables.FONT_WEIGHT.DEMI_BOLD};
-    font-size: ${variables.FONT_SIZE.SMALL};
-    padding: 0 32px 0 20px;
-    color: ${({ theme }) => theme.TYPE_DARK_GREY};
-    background: transparent;
-    box-sizing: border-box;
+    input {
+        letter-spacing: 8px;
```
Comment (AdamSchinzel) [inline packages/suite/src/components/suite/PinInput/components/InputPin/index.tsx]: Can be used from `spacingsPx`

File: packages/suite/src/components/suite/PinInput/components/InputPin/index.tsx [hunk 7]
```diff
@@ -3,15 +3,17 @@ import styled from 'styled-components';
...
+        min-width: 256px;
+        height: 54px;
+        font-weight: ${variables.FONT_WEIGHT.DEMI_BOLD};
+        font-size: ${variables.FONT_SIZE.SMALL};
+        padding: 0 32px 0 20px;
```
Comment (AdamSchinzel) [inline packages/suite/src/components/suite/PinInput/components/InputPin/index.tsx]: Can be used from `spacingsPx`

File: packages/suite/src/views/wallet/transactions/TransactionList/TransactionListActions/SearchAction.tsx [hunk 8]
```diff
@@ -35,13 +35,15 @@ const Wrapper = styled.div<{ expanded: boolean }>`
...
-            cursor: pointer;
-            padding: 0 16px;
-        `}
+    input {
+        border: none;
+        ${({ expanded }) =>
+            !expanded &&
+            css`
+                cursor: pointer;
+                padding: 0 16px;
```
Comment (AdamSchinzel) [inline packages/suite/src/views/wallet/transactions/TransactionList/TransactionListActions/SearchAction.tsx]: Can be used from `spacingsPx`

**Review by AdamSchinzel — COMMENTED** [review]
Just places where I can think we can use `spacingsPx`

File: packages/components/src/components/form/SelectBar/SelectBar.tsx (hunk 5 above)
Comment (AdamSchinzel) [reply packages/components/src/components/form/SelectBar/SelectBar.tsx:30]: Ok, just added last stuff regarding this

File: packages/suite/src/components/suite/PinInput/components/InputPin/index.tsx (hunk 6 above)
Comment (dahaca) [reply packages/suite/src/components/suite/PinInput/components/InputPin/index.tsx]: cdd1b94ab

But, please, do not concentrate on it outside of the `components` package for now 🙌

File: packages/suite/src/components/suite/PinInput/components/InputPin/index.tsx (hunk 7 above)
Comment (dahaca) [reply packages/suite/src/components/suite/PinInput/components/InputPin/index.tsx]: cdd1b94ab

File: packages/suite/src/views/wallet/transactions/TransactionList/TransactionListActions/SearchAction.tsx (hunk 8 above)
Comment (dahaca) [reply packages/suite/src/views/wallet/transactions/TransactionList/TransactionListActions/SearchAction.tsx]: cdd1b94ab

**Review by AdamSchinzel — APPROVED** [review]
lgtm
